src/api/XR.js

```diff
@@ -80,10 +102,42 @@ export default class XR extends EventTarget {
       }
     }
 
+    if (!XRSessionModes.includes(mode)) {
+      throw new TypeError(
+          `The provided value '${mode}' is not a valid enum value of type XRSessionMode`);
+    }
+
+    // Resolve which of the requested features are supported and reject if a
+    // required feature isn't available.
+    const requestedOptions = Object.assign({}, DEFAULT_SESSION_OPTIONS[mode], options);
```

Comment:
Two things to note here, although not sure if it's an issue:

* The arrays of required and optional features in `requestedOptions` will be the same instance as the default or the passed in options -- these are rebuilt as a set (`enabledFeatures`) which will eliminate that chance, however
* I don't know the details of the spec in this regard, but if any required or optional features are declared, it will completely override the defaults, e.g. `{ requiredFeatures: ['foo'] }` for an inline session would remove the default `{ requiredFeatures: ['viewer'] }` viewer feature